### PR TITLE
#1 - P2P Sessions

### PR DESCRIPTION
**Pull request by denis-onder**
Issue #1

TL;DR - What does this implement?
---------------------------------
Satisfies the requirements of #1.
Established a P2P stream between two peers using the provided test client.

How to Test
-----------
Once the Node server has been started, navigate to `https://locahost:<PORT>/test` and start a stream. After that, navigate to `https://localhost:<PORT>/sessions/ignacius` and enjoy the stream!

Worries / Concerns
------------------
Cross-browser compactibility and video playback quality.

Checklist
---------

 - [ ] Do we need to test this?
 - [ ] Were tests written?
 - [x] Does this meet coding standards?
 - [x] Is this is code change?
 - [ ] Is this a meta change (e.g issue template)?
 - [x] Is this pull request ready for review?
 - [ ] Is this a breaking change (e.g not backwards compatible with clients)?
 
ESLint Output
-------------

```                                                  
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.
```

Reviewers
---------

(Tag people here)
@gmemstr

### CODE REVIEW

File: app/test-client/index.html [hunk 1]
```diff
@@ -4,14 +4,50 @@
...
+    <button id="share">Share screen</button>
     <script src="https://unpkg.com/peerjs@1.0.0/dist/peerjs.min.js"></script>
+    <script src="https://webrtc.github.io/adapter/adapter-latest.js"></script>
+    <script src="https://cdn.firebase.com/v0/firebase.js"></script>
+    <script src="https://cdn.webrtc-experiment.com/getScreenId.js"></script>
+    <script src="https://cdn.webrtc-experiment.com/screen.js"></script>
     <script>
-      const peer = new Peer({ host: 'localhost', port: 8000 });
-      peer.connect();
+      const peer = new Peer('ignacius', { host: 'localhost', port: 8000 });
```
Comment (gmemstr) [inline app/test-client/index.html]: Shouldn't be hardcoded, or should at least be on the default port?

File: scripts/generateSSL.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+fi
+
+cd ./keys
+
+rm *domain*
+
+echo Generating SSL Key/Cert pair...
+
+openssl req \
+  -newkey rsa:2048 -nodes -keyout domain.key \
```
Comment (gmemstr) [inline scripts/generateSSL.sh]: Should be completely automatic and not prompt for any entry, see https://unix.stackexchange.com/questions/104171/create-ssl-certificate-non-interactively

File: app/test-client/index.html (hunk 1 above)
Comment (gmemstr) [reply app/test-client/index.html]: `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://localhost:8000/peerjs/id?ts=15714830210200.6787841462309826. (Reason: CORS request did not succeed).`

Comment (denis-onder) [reply app/test-client/index.html]: This is hard-coded for testing purposes. If required, I can change this to be dynamic, but I don't see the point since this won't be available in any other environment except the development one.

Comment (gmemstr) [reply app/test-client/index.html]: Should at least document it for dev purposes :)

Comment (denis-onder) [reply app/test-client/index.html]: Documented!

File: scripts/generateSSL.sh (hunk 2 above)
Comment (denis-onder) [reply scripts/generateSSL.sh]: Tested, and implemented a script from this post. Works as intended, completely automatic!
